js/ui/calendar/ui.calendar.selection.strategy.js

```diff
@@ -0,0 +1,227 @@
+import { noop } from '../../core/utils/common';
+import Class from '../../core/class';
+import dateUtils from '../../core/utils/date';
+
+const abstract = Class.abstract;
+
+const DAY_INTERVAL = 86400000;
+
+const CalendarSelectionStrategy = Class.inherit({
+    ctor: function(calendar) {
+        this.calendar = calendar;
+    },
+
+    getViewOptions: abstract,
+
+    dateOption: function(optionName) {
+        return this.calendar._dateOption(optionName);
+    },
+
+    dateValue: function(value, e) {
+        this.calendar._dateValue(value, e);
+    },
+
+    updateAriaSelected: function(value, previousValue) {
+        this.calendar._updateAriaSelected(value, previousValue);
+
+        if(value[0] && this.calendar.option('currentDate').getTime() === value[0].getTime()) {
+            this.calendar._updateAriaId(value[0]);
+        }
+    },
+
+    processValueChanged: function(value, previousValue) {
+        value = (value || []).map((item) => this._convertToDate(item));
+        previousValue = (previousValue || []).map((item) => this._convertToDate(item));
+
+        this._updateViewsValue(value);
+        this.updateAriaSelected(value, previousValue);
+
+        !this._currentDateChanged && this.calendar._initCurrentDate();
```

Comment:
It's not a good practice to use logical operator instead of usual statement
https://github.com/airbnb/javascript#control-statements--value-selection
```suggestion
        if(!this._currentDateChanged) {
            this.calendar._initCurrentDate();
        }
```

Reply:
done